### PR TITLE
Fix put_object_copy headers

### PR DESCRIPTION
**Pull request by ericmj**

### CODE REVIEW

File: lib/ex_aws/s3/client.ex [hunk 1]
```diff
@@ -312,15 +313,24 @@ defmodule ExAws.S3.Client do
...
+    | {:cache_control, binary}
+    | {:content_disposition, binary}
+    | {:content_encoding, binary}
+    | {:content_length, binary}
+    | {:content_type, binary}
+    | {:expect, binary}
+    | {:expires, binary}
+    | {:storage_class, :standard | :redunced_redundancy}
+    | {:website_redirect_location, binary}
+    | {:meta, amz_meta_opts}
```
Comment (benwilson512) [inline lib/ex_aws/s3/client.ex]: This is awesome, thank you!

File: lib/ex_aws/s3/client.ex [hunk 2]
```diff
@@ -312,15 +313,24 @@ defmodule ExAws.S3.Client do
   defcallback put_object_acl!(bucket :: binary, object :: binary, acl :: acl_opts) :: ExAws.Request.success_content
 
   @type pub_object_copy_opts :: [
-    {:metadata_directive, :copy | :replace}
+    {:metadata_directive, :COPY | :REPLACE}
     | {:copy_source_if_modified_since, binary}
     | {:copy_source_if_unmodified_since, binary}
     | {:copy_source_if_match, binary}
     | {:copy_source_if_none_match, binary}
-    | {:storage_class, :standard, :redunced_redundancy}
```
Comment (benwilson512) [inline lib/ex_aws/s3/client.ex]: good catch.

File: lib/ex_aws/s3/client.ex [hunk 3]
```diff
@@ -312,15 +313,24 @@ defmodule ExAws.S3.Client do
   defcallback put_object_acl!(bucket :: binary, object :: binary, acl :: acl_opts) :: ExAws.Request.success_content
 
   @type pub_object_copy_opts :: [
-    {:metadata_directive, :copy | :replace}
+    {:metadata_directive, :COPY | :REPLACE}
```
Comment (ericmj) [inline lib/ex_aws/s3/client.ex]: Not sure if you'd rather downcase these?

Comment (benwilson512) [inline lib/ex_aws/s3/client.ex]: aside from `:DOWN` and a few others `:FOO` style atoms aren't super common. Leaving them downcase means I need to capitalize them in ex_aws I assume yes?

File: lib/ex_aws/s3/impl.ex [hunk 4]
```diff
@@ -351,16 +351,14 @@ defmodule ExAws.S3.Impl do
     |> Map.get(:destination_encryption, %{})
     |> build_encryption_headers
 
-    meta = opts
-    |> Map.get(:meta, %{})
-    |> build_meta_headers
+    regular_headers = opts
+    |> Map.delete(:encryption)
```
Comment (benwilson512) [inline lib/ex_aws/s3/impl.ex]: Why is this necessary?

File: lib/ex_aws/s3/client.ex (hunk 3 above)
Comment (ericmj) [inline lib/ex_aws/s3/client.ex]: Correct.

File: lib/ex_aws/s3/impl.ex (hunk 4 above)
Comment (ericmj) [inline lib/ex_aws/s3/impl.ex]: It's not strictly necessary but copy uses `:source_encryption` and `: destination_encryption` instead.

Comment (benwilson512) [inline lib/ex_aws/s3/impl.ex]: Yeah fair. And the other items are already handled by `put_object_headers`
